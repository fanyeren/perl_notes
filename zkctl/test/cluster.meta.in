#!/bin/bash
./zkctl -s cluster repair
./zkctl -w cluster set-meta-all --data={}

raw='"bs_deploy_dyndata_script_bandwidth_limit":"10240","bs_deploy_dyndata_script_timeout":"100","bs_deploy_script_bandwidth_limit":"10240","bs_deploy_script_timeout":"600","bs_healthcheck_script_timeout":"10","bs_reload_script_timeout":"30","bs_start_script_timeout":"300","bs_stop_script_timeout":"30","bs_switchdb_script_timeout":"10","bs_transdb_script_bandwidth_limit":"10240","bs_transdb_script_timeout":"3600"'

for item in `echo $raw|tr -s "," " "`; do
    rawkv=`echo $item|tr -d \" | tr -s ":" " "`
    kv=($rawkv)
    ./zkctl -w cluster add-meta-attr --key=${kv[0]} --value=${kv[1]}
    echo "Add meta ${kv[0]}=${kv[1]} result $?"
    ./zkctl -w cluster get-meta-attr --key=${kv[0]}
    echo "Get meta ${kv[0]}=${kv[1]} result $?"
done

./zkctl -s cluster get-meta-all
echo "got all cluster result result=$?"

for item in `echo $raw|tr -s "," " "`; do
    rawkv=`echo $item|tr -d \" | tr -s ":" " "`
    kv=($rawkv)
    ./zkctl -w cluster del-meta-attr --key=${kv[0]}
    echo "del meta attr ${kv[0]} result=$?"
done

./zkctl -s cluster get-meta-all
echo "got empty cluster result result=$?"
