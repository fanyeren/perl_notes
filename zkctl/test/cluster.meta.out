Add meta bs_deploy_dyndata_script_bandwidth_limit=10240 result 0
10240
Get meta bs_deploy_dyndata_script_bandwidth_limit=10240 result 0
Add meta bs_deploy_dyndata_script_timeout=100 result 0
100
Get meta bs_deploy_dyndata_script_timeout=100 result 0
Add meta bs_deploy_script_bandwidth_limit=10240 result 0
10240
Get meta bs_deploy_script_bandwidth_limit=10240 result 0
Add meta bs_deploy_script_timeout=600 result 0
600
Get meta bs_deploy_script_timeout=600 result 0
Add meta bs_healthcheck_script_timeout=10 result 0
10
Get meta bs_healthcheck_script_timeout=10 result 0
Add meta bs_reload_script_timeout=30 result 0
30
Get meta bs_reload_script_timeout=30 result 0
Add meta bs_start_script_timeout=300 result 0
300
Get meta bs_start_script_timeout=300 result 0
Add meta bs_stop_script_timeout=30 result 0
30
Get meta bs_stop_script_timeout=30 result 0
Add meta bs_switchdb_script_timeout=10 result 0
10
Get meta bs_switchdb_script_timeout=10 result 0
Add meta bs_transdb_script_bandwidth_limit=10240 result 0
10240
Get meta bs_transdb_script_bandwidth_limit=10240 result 0
Add meta bs_transdb_script_timeout=3600 result 0
3600
Get meta bs_transdb_script_timeout=3600 result 0
{"bs_deploy_dyndata_script_bandwidth_limit":"10240","bs_deploy_dyndata_script_timeout":"100","bs_deploy_script_bandwidth_limit":"10240","bs_deploy_script_timeout":"600","bs_healthcheck_script_timeout":"10","bs_reload_script_timeout":"30","bs_start_script_timeout":"300","bs_stop_script_timeout":"30","bs_switchdb_script_timeout":"10","bs_transdb_script_bandwidth_limit":"10240","bs_transdb_script_timeout":"3600"}
got all cluster result result=0
del meta attr bs_deploy_dyndata_script_bandwidth_limit result=0
del meta attr bs_deploy_dyndata_script_timeout result=0
del meta attr bs_deploy_script_bandwidth_limit result=0
del meta attr bs_deploy_script_timeout result=0
del meta attr bs_healthcheck_script_timeout result=0
del meta attr bs_reload_script_timeout result=0
del meta attr bs_start_script_timeout result=0
del meta attr bs_stop_script_timeout result=0
del meta attr bs_switchdb_script_timeout result=0
del meta attr bs_transdb_script_bandwidth_limit result=0
del meta attr bs_transdb_script_timeout result=0
{}
got empty cluster result result=0
